src/game/object/blocks.ts

```diff
@@ -27,42 +29,34 @@ function convertElementToBlock(element: Element): Block {
     // e.g. http://abehiroshi.la.coocan.jp/
     uuid: Math.random().toString(36).slice(-8),
     element,
-    rect: getRectOfBlock(element),
+    rects: getRectsOfBlock(element),
     remain: true,
     removed: false
   }
 }
 
-export function getRectOfBlock(element: Element) {
-  const __rect = element.getBoundingClientRect()
-  const _rect = {
-    top: __rect.top,
-    bottom: __rect.bottom,
-    left: __rect.left,
-    right: __rect.right
-  }
-
-  const offsetAddedRect = addFrameOffsetToRect(element, _rect)
-
-  const rect = {
-    top: window.innerHeight - offsetAddedRect.top,
-    bottom: window.innerHeight - offsetAddedRect.bottom,
-    left: offsetAddedRect.left,
-    right: offsetAddedRect.right
+export function getRectsOfBlock(element: Element): Rect[] {
+  const result: Rect[] = [];
+  const rects = element.getClientRects();
+  for (let i = 0; i < rects.length; i++) {
+    const { top, bottom, left, right } = rects[i];
+    const rect = { top, bottom, left, right }
+    const offsetAddedRect = addFrameOffsetToRect(element, rect)
+    result.push({
+      top: window.innerHeight - offsetAddedRect.top,
+      bottom: window.innerHeight - offsetAddedRect.bottom,
+      left: offsetAddedRect.left,
+      right: offsetAddedRect.right
+    });
   }
-
-  return rect
+  return result;
 }
 
 function addFrameOffsetToRect(
   element: Element,
-  rect: {
-    top: number
-    bottom: number
-    left: number
-    right: number
-  }
-) {
+  rect: Rect
+): Rect {
+  let result = { ...rect };
```

Comment:
```suggestion
  const result = { ...rect };
```
Properties of 'const' objects can be changed, so I think you can use 'const' here :)

Reply:
If you wanna use 'let' as a marker for the usage of while loop, is the below also an idea?
```suggestion
  let { top, bottom, left, right } = rect
```

Reply:
Nice catch! I'll adopt the former.

Reply:
thanks!